### ISSUE TITLE
NETStandard 1.6 OOB compatibility issues

### ISSUE THREAD

**Issue by Silvenga** [paulcbetts/refit#259]
NETStandard 1.6 breaks compatibility with several libraries as it depends on the OOB System.Net 4.1.0.0. One major library impacted is Asp.Net Web API 2 as it depends on 4.0.0.0 included with the .Net Framework. 

```
[VerificationException: Method System.Net.Http.CloneableExtensions.Clone: type argument 'System.Net.Http.Headers.MediaTypeHeaderValue' violates the constraint of type parameter 'T'.]
```

The issue is resolved and is currently pending release as 1.1 in the Fall.

https://github.com/dotnet/corefx/issues/9846
https://github.com/dotnet/corefx/issues/9884

If impacted, my proposal is to downgrade refit until the CoreFX team releases 1.1. See https://github.com/AutoMapper/AutoMapper/issues/1526.

**Comment by onovotny** [comment]
Can't downgrade refit because then it won't work on .NET Core libraries. There are two workarounds:

1. Put a binding redirect to use System.Net.Http 4.0.0 (that works for me on desktop .net)
2. Stick with Refit 2.5 as Netstandard.library is really the only change for 3.0

**Comment by Silvenga** [comment]
Sorry, I forgot to mention that this issue is entirely for documentation for others, since I had to spend some time sifting through CoreFX issues to find the root reason (maybe save others time). I didn't think it's practical to downgrade refit libraries for an external issue, rather we should downgrade refit if we are impacted (as you suggested). 

> Put a binding redirect to use System.Net.Http 4.0.0 (that works for me on desktop .net)

Does refit depend on anything in System.Net.Http 4.1? It also feels odd to include a reference to  System.Net.Http 4.1 and include it in the deployment if we are truly running with the GAC 4.0. :frowning: Although, we could also replace the reference to 4.1 with 4.0 which would prevent output pollution, but messing with references added by Nuget also feels wrong.

**Comment by onovotny** [comment]
This seems fixed in the CI feeds of Core FX https://github.com/dotnet/corefx/issues/11100

**Comment by onovotny** [comment]
Solution here is to use the 4.3.1+ package of System.Net.Http. Refit can automatically pull this in for the next release.

https://github.com/dotnet/corefx/issues/11100#issuecomment-301601156

**Comment by onovotny** [comment]
Will be fixed in 3.1
